webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/nodes/NodeGroup.scala

```diff
@@ -37,15 +37,165 @@
 
 package com.normation.rudder.domain.nodes
 
+import com.normation.errors.Inconsistancy
+import com.normation.errors.RudderError
+import com.normation.errors.SystemError
 import com.normation.inventory.domain.NodeId
+import com.normation.rudder.domain.queries.And
+import com.normation.rudder.domain.queries.CriterionLine
 import com.normation.rudder.domain.queries.Query
+import com.normation.rudder.domain.queries.SubGroupComparator
+import net.liftweb.json.DefaultFormats
+import net.liftweb.json.JArray
+import net.liftweb.json.JField
+import net.liftweb.json.JObject
+import net.liftweb.json.JsonAST.JString
+import net.liftweb.json.JsonAST.JValue
+import net.liftweb.json._
+import net.liftweb.json.JsonDSL._
+import net.liftweb.json.JsonParser.ParseException
 
 /**
  * UUId type for Node Groups, so that they
  * can be uniquely identified in our world.
  */
 final case class NodeGroupId(value:String) extends AnyVal
 
+final case class GroupProperty(
+    name    : String
+  , value   : JValue
+) {
+  def renderValue: String = value match {
```

Comment:
Maybe this could be in GenericPropertyUtils

Or having a trait for Properties having this type and then a function in this trait or an implicit class providing this function

Reply:
I thought a bit to that, but I went to `GroupProperties` because:
- `NodeProperties` are reasonably different, both in code (because of provider) and in semantic (because the overriding is totally different, they are the only one that matters in the end: they are the only one to go to node
- I'm not sure that if we add other properties, they will be the same.

But in current way of doing things, it's rather bad. We have double serialisation for each of them, which mean a change to define incompatible format and make impossible sharing things like the property editor. 
So I think it should be done, not sure how (perhaps a `property` which is just a `name` + `value`, and a `case class NodeProperty(property: Property, provider: ...)` and smart constructor/extractors...)

Reply:
I think having an inner Property with common behavior for all properties would be interesting, but maybe we can keep it for another PR

Reply:
Yes, we can. And I'm not sure we can makes much more than what is in `GenericPropertyUtils`, because for three cases (global param, node prop, group prop), we have 3 differents way of serializing (totally different for param, and node has more info than group). Perhaps there's something that can be done regarding event log, and even there, we have similar problems